dali/python/nvidia/dali/external_source.py

```diff
@@ -149,6 +194,16 @@ def _cycle_enabled(cycle):
   - "quiet", True - quietly rewind the data
   - "raise" - raise StopIteration on each rewind.""".format(repr(cycle)))
 
+def _accepted_arg_count(callable):
+    if not inspect.isfunction(callable) and not inspect.ismethod(callable) and hasattr(callable, '__call__'):
+        callable = callable.__call__
+    if not inspect.ismethod(callable):
+        implicit_args = 0
+    else:
+        implicit_args = 1
+        callable = callable.__func__
```

Comment:
```suggestion
    if not inspect.ismethod(callable):
        implicit_args = 0
        if not inspect.isfunction(callable) and hasattr(callable, '__call__'):
           callable = callable.__call__
    else:
        implicit_args = 1
        callable = callable.__func__
```

Reply:
I no longer can see what is happening here or why should I change. Will revisit later. Can you motivate why one over the other?

Reply:
To not repeat `not inspect.ismethod(callable)` twice.

Reply:
But that changes the order of the operations, if we were passed an object with `__call__`, we extract the `__call__` and next check if this extracted callable `__call__` is function or method (in the case of getting an object to call, it probably means we will call it's `__call__` method, so it has 1 implicit `self` argument).

Reply:
True. Missed that.